### DISCUSSION TITLE
Vite Plugins API Handler

### DISCUSSION THREAD

**Discussion by Ponjimon** [TanStack/router#2428] [Q&A]
https://github.com/TanStack/router/blob/a42deba77679f221d688eef5e6f121d7bc00731b/packages/start/src/config/index.ts#L317

Is there a reason the API handler shares its Vite plugins with the ones for the SSR handler? 

I'm trying to find a way to make Cloudflare Bindings work locally. For that, I adapted the `cloudflareDevProxyVitePlugin` that Remix uses and basically inject the Wrangler Context into the H3Event context.

But because the plugins are shared, once I use that plugin, while the API routes still work, the regular routes stop working and throw a 404 error. Presumably, because I override the SSR handler with my plugin.

**Comment by SeanCassiere** [comment]
Please create an issue for this.

This won't be difficult to implement. Feel free to give it a shot if you've got the time. It'll just be extending the `routers` in the config to support API and its own vite keys.

**Reply by SeanCassiere** [reply]
Changes happening
- https://github.com/TanStack/router/blob/d9ea07d7ef77111ef90a2053617b9ff41267e026/packages/start/src/config/index.ts#L193
- https://github.com/TanStack/router/blob/d9ea07d7ef77111ef90a2053617b9ff41267e026/packages/start/src/config/index.ts#L317

**Reply by Ponjimon** [reply]
Thanks, that's what I thought. I just wasn't sure if this was intended or maybe a mistake! Will create a PR!

**Comment by Ponjimon** [comment]
PR has been created: https://github.com/TanStack/router/pull/2436
